File: frontend/src/services/attendanceService.ts
import api from './api'
import { Attendance } from '@/types'

export const attendanceService = {
  async markAttendance(attendance: Partial<Attendance>): Promise<Attendance> {
    const response = await api.post<Attendance>('/teacher/attendance', attendance)
    return response.data
  },

  async updateAttendance(id: number, attendance: Partial<Attendance>): Promise<Attendance> {
    const response = await api.put<Attendance>(`/teacher/attendance/${id}`, attendance)
    return response.data
  },

  async getAttendanceByStudent(studentId: number): Promise<Attendance[]> {
    const response = await api.get<Attendance[]>(`/teacher/attendance/student/${studentId}`)
    return response.data
  },

  async getAttendanceByCourse(courseId: number): Promise<Attendance[]> {
    const response = await api.get<Attendance[]>(`/teacher/attendance/course/${courseId}`)
    return response.data
  },
}

File: src/main/java/com/eadms/service/AttendanceService.java
package com.eadms.service;

import com.eadms.dto.request.AttendanceEntryRequest;
import com.eadms.dto.response.AttendanceResponse;

import java.util.List;
import java.util.Map;

public interface AttendanceService {
    AttendanceResponse markAttendance(AttendanceEntryRequest request);
    AttendanceResponse updateAttendance(Long attendanceId, AttendanceEntryRequest request);
    List<AttendanceResponse> getAttendanceByStudent(Long studentId);
    List<AttendanceResponse> getAttendanceByCourse(Long courseId);
    Double calculateAttendancePercentage(Long studentId, Long courseId);
    Map<String, Object> getAttendanceStats(Long studentId);
}
